src/hook_manager.cc

```diff
@@ -123,6 +123,17 @@ void HookManager::run_hook(StringView hook_name, StringView param, Context& cont
 
             CommandManager::instance().execute(to_run.hook->commands, context,
                                                { {}, std::move(env_vars) });
+
+            if (to_run.hook->flags & HookFlags::Once)
+            {
+                auto it = std::remove_if(hook_list->value.begin(), hook_list->value.end(),
+                                       [&](const std::unique_ptr<Hook>& h)
+                                       { return h.get() == to_run.hook; });
+
+                m_hooks_trash.insert(m_hooks_trash.end(), std::make_move_iterator(it),
+                                     std::make_move_iterator(hook_list->value.end()));
+                hook_list->value.erase(it);
+            }
```

Comment:
Please pay extra attention to this hunk, it seems to work but I'm not confident it's what's needed to properly remove a hook from the hooks list.

Reply:
This is a strange way to do it, it seems correct, but is more complex than necessary.

* `std::remove_if` will move all the hooks to remove at the end of the container and return an iterator to the first one of those "removed" elements. It is strange to use it because we should have exactly one element to remove, so just using `find_if(hook_list->value, ...);` should be enough. 
* m_hooks_trash.insert is unnecessary, use push_back or emplace_back, as we have a single element, and just use `std::move(*it)` to move the hook out.

Reply:
I originally did use `find_if()` and a simple `push_back(std::move(*it));`, but then got weird compilation errors (because I didn't notice straight away this function was `const`!), so I based this bit on `remove_hooks()`.

I'll revert the changes back.